easypermissions/src/main/java/pub/devrel/easypermissions/EasyPermissions.java

```diff
@@ -247,8 +257,18 @@ private static void runAnnotatedMethods(Object object, int requestCode) {
 
     private static void checkCallingObjectSuitability(Object object) {
         // Make sure Object is an Activity or Fragment
-        if (!((object instanceof Fragment) || (object instanceof Activity))) {
-            throw new IllegalArgumentException("Caller must be an Activity or a Fragment.");
+        boolean isActivity = object instanceof Activity;
+        boolean isSupportFragment = object instanceof Fragment;
+        boolean isAppFragment = object instanceof android.app.Fragment;
+
+        if (!(isSupportFragment || isActivity ||
+                (isAppFragment && android.os.Build.VERSION.SDK_INT >= 23))) {
```

Comment:
Thanks!  Just the one issue below remaining, which I justified with a link to a possible edge case.  Thanks for the update.